src/connect.rs

```diff
@@ -160,9 +172,35 @@ impl Connect for Connector {
                     #[cfg(not(feature = "tls"))]
                     Inner::Http(http) => connect!(http, $dst, $proxy),
                     #[cfg(feature = "default-tls")]
-                    Inner::DefaultTls(http, _) => connect!(http, $dst, $proxy),
+                    Inner::DefaultTls(http, tls) => {
+                        let mut http = http.clone();
+                        http.set_nodelay(nodelay);
+                        let http = ::hyper_tls::HttpsConnector::from((http, tls.clone()));
+                        connect!(http, $dst, $proxy)
+                    },
                     #[cfg(feature = "rustls-tls")]
-                    Inner::RustlsTls(http, _) => connect!(http, $dst, $proxy)
+                    Inner::RustlsTls { http, tls, .. } => {
+                        let mut http = http.clone();
+
+                        // Disable Nagle's algorithm for TLS handshake
+                        //
+                        // https://www.openssl.org/docs/man1.1.1/man3/SSL_connect.html#NOTES
+                        http.set_nodelay(nodelay || ($dst.scheme() == "https"));
```

Comment:
Do you think this complexity should be kept around now that is possible to disable Nagle's?

Reply:
I think disable Nagle's for TLS handshake is good in most cases.